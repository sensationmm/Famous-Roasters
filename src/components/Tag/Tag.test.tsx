import { render, waitFor } from '@testing-library/react'
import React from 'react'

import { Tag, TagType } from '.'

describe('Tag component', () => {
  it('Renders correctly', async () => {
    const { container } = render(<Tag value="sweet" />)
    await waitFor(() => new Promise((res) => setTimeout(res, 0)))
    expect(container).toMatchSnapshot()
  })

  it('Renders correctly for aroma tag with value experimentell & komplex', async () => {
    const { container } = render(<Tag type={TagType.Aroma} value="experimentell & komplex" />)
    await waitFor(() => new Promise((res) => setTimeout(res, 0)))
    expect(container).toMatchSnapshot()
  })

  it('Renders correctly for aroma tag with value fruchtig & lebhaft', async () => {
    const { container } = render(<Tag type={TagType.Aroma} value="fruchtig & lebhaft" />)
    await waitFor(() => new Promise((res) => setTimeout(res, 0)))
    expect(container).toMatchSnapshot()
  })

  it('Renders correctly for aroma tag with value floral & leicht', async () => {
    const { container } = render(<Tag type={TagType.Aroma} value="floral & leicht" />)
    await waitFor(() => new Promise((res) => setTimeout(res, 0)))
    expect(container).toMatchSnapshot()
  })

  it('Renders correctly for aroma tag with value nussig & schokoladig', async () => {
    const { container } = render(<Tag type={TagType.Aroma} value="nussig & schokoladig" />)
    await waitFor(() => new Promise((res) => setTimeout(res, 0)))
    expect(container).toMatchSnapshot()
  })

  it('Renders correctly for aroma tag with value w체rzig & kr채ftig', async () => {
    const { container } = render(<Tag type={TagType.Aroma} value="w체rzig & kr채ftig" />)
    await waitFor(() => new Promise((res) => setTimeout(res, 0)))
    expect(container).toMatchSnapshot()
  })

  it('Renders correctly for aroma tag with a different value', async () => {
    const { container } = render(<Tag type={TagType.Aroma} value="something else" />)
    await waitFor(() => new Promise((res) => setTimeout(res, 0)))
    expect(container).toMatchSnapshot()
  })

  it('Renders correctly for type taste finder', async () => {
    const { container } = render(<Tag type={TagType.TasteFinder} value="100% matching" />)
    await waitFor(() => new Promise((res) => setTimeout(res, 0)))
    expect(container).toMatchSnapshot()
  })
})
