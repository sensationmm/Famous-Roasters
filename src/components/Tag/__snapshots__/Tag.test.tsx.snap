// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tag component Renders correctly 1`] = `
<div>
  <div
    class="inline-flex rounded-full px-3 py-1 bg-brand-grey-bombay"
  >
    <span
      class="text-xs leading-4"
    >
      sweet
    </span>
  </div>
</div>
`;

exports[`Tag component Renders correctly for aroma tag with a different value 1`] = `
<div>
  <div
    class="inline-flex rounded-full px-3 py-1 bg-brand-grey-bombay"
  >
    <span
      class="text-xs leading-4"
    >
      something else
    </span>
  </div>
</div>
`;

exports[`Tag component Renders correctly for aroma tag with value experimentell & komplex 1`] = `
<div>
  <div
    class="inline-flex rounded-full px-3 py-1 bg-tags-capeHoney"
  >
    <span
      class="text-xs leading-4"
    >
      experimentell & komplex
    </span>
  </div>
</div>
`;

exports[`Tag component Renders correctly for aroma tag with value floral & leicht 1`] = `
<div>
  <div
    class="inline-flex rounded-full px-3 py-1 bg-tags-melrose"
  >
    <span
      class="text-xs leading-4"
    >
      floral & leicht
    </span>
  </div>
</div>
`;

exports[`Tag component Renders correctly for aroma tag with value fruchtig & lebhaft 1`] = `
<div>
  <div
    class="inline-flex rounded-full px-3 py-1 bg-tags-cottonCandy"
  >
    <span
      class="text-xs leading-4"
    >
      fruchtig & lebhaft
    </span>
  </div>
</div>
`;

exports[`Tag component Renders correctly for aroma tag with value nussig & schokoladig 1`] = `
<div>
  <div
    class="inline-flex rounded-full px-3 py-1 bg-tags-coldTurkey"
  >
    <span
      class="text-xs leading-4"
    >
      nussig & schokoladig
    </span>
  </div>
</div>
`;

exports[`Tag component Renders correctly for aroma tag with value w체rzig & kr채ftig 1`] = `
<div>
  <div
    class="inline-flex rounded-full px-3 py-1 bg-tags-yourPink"
  >
    <span
      class="text-xs leading-4"
    >
      w체rzig & kr채ftig
    </span>
  </div>
</div>
`;

exports[`Tag component Renders correctly for type taste finder 1`] = `
<div>
  <div
    class="inline-flex rounded-full px-3 py-1 bg-brand-blue text-tags-jacksonsBlue"
  >
    <span
      class="text-xs leading-4"
    >
      100% matching
    </span>
  </div>
</div>
`;

exports[`TagSwatch component Renders correctly 1`] = `
<div>
  <span
    class="inline-flex rounded-full px-3 py-1 bg-brand-grey-bombay w-6 h-6 px-0 py-0 rounded-3xl mr-3 flex-shrink-0"
  />
</div>
`;
